assets/src/edit-story/elements/text/display.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+/**
+ * WordPress dependencies
+ */
+import { useRef, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import getCaretCharacterOffsetWithin from '../../utils/getCaretCharacterOffsetWithin';
+import { useStory } from '../../app';
+import { useCanvas } from '../../components/canvas';
+import {
+	ElementWithPosition,
+	ElementWithSize,
+	ElementWithFont,
+	ElementWithBackgroundColor,
+	ElementWithFontColor,
+} from '../shared';
+
+const Element = styled.p`
+	margin: 0;
+	${ ElementWithPosition }
+	${ ElementWithSize }
+	${ ElementWithFont }
+	${ ElementWithBackgroundColor }
+	${ ElementWithFontColor }
+
+	user-select: ${ ( { canSelect } ) => canSelect ? 'initial' : 'none' };
+
+	&:focus {
+		outline: none;
+	}
+`;
+
+function TextDisplay( { id, content, color, backgroundColor, width, height, x, y, fontFamily, fontSize, fontWeight, fontStyle } ) {
+	const props = {
+		color,
+		backgroundColor,
+		fontFamily,
+		fontStyle,
+		fontSize,
+		fontWeight,
+		width,
+		height,
+		x,
+		y,
+	};
+	const {
+		state: { selectedElementIds },
+	} = useStory();
+	const {
+		actions: { setEditingElement, setEditingElementWithState },
+	} = useCanvas();
+	const isElementSelected = selectedElementIds.includes( id );
+	const isElementOnlySelection = isElementSelected && selectedElementIds.length === 1;
+	const handleClick = ( evt ) => {
+		if ( evt.shiftKey || evt.metaKey || evt.altKey || evt.ctrlKey ) {
+			// Some modifier was pressed. Ignore and bubble
+			return;
+		}
+		// Enter editing without and place cursor at current selection offset
+		setEditingElementWithState( id, { offset: getCaretCharacterOffsetWithin( element.current ) } );
+		evt.stopPropagation();
+	};
+	const handleKeyDown = ( evt ) => {
+		if ( evt.metaKey || evt.altKey || evt.ctrlKey ) {
+			// Some modifier (except shift) was pressed. Ignore and bubble
+			return;
+		}
+
+		if ( evt.key === 'Enter' ) {
```

Comment:
Isn't better to do a key code check here like [this](https://github.com/ampproject/amp-wp/blob/8ae3c05f0f525ed85305527777d5ef07d40a3ae0/assets/src/stories-editor/components/draggable-text.js#L68)

Reply:
[`keyCode` is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode).